db/fetcher/fetcher.go

```diff
@@ -118,14 +149,29 @@ func (df *DocumentFetcher) init(
 	}
 	df.kvIter = nil
 
-	df.schemaFields = make(map[uint32]client.FieldDescription)
-	for _, field := range col.Schema.Fields {
-		df.schemaFields[uint32(field.ID)] = field
+	df.selectFields = make(map[uint32]client.FieldDescription, len(fields))
+	// @todo: We can probably make this size more accurate, this is an upper bound
```

Comment:
question: How would you make this more accurate? You seem to be using all the allocated KVs in the for loop.

Reply:
the `@todo` here is old, the line it refers to has been removed since opening the PR xD